### PR TITLE
Fixes SI-5593

### PR DESCRIPTION
**Pull request by VladUreche**
The problem was the classpath was sneakily set to "." whenever no
classpath was given. Paul's comment says it all:

/*\* [...]
 *
-  - If -classpath or -cp is given, it is that
-  - Otherwise, if CLASSPATH is set, it is that
-  - If neither of those, then "." is used.
  */

Fixed for Scaladoc, if the classpath is empty, it stays empty. Scalac
still has the "." fallback classpath, as before, I'm not touching that.

### CODE REVIEW

**Comment by paulp** [comment]
No, this doesn't work.  This area is very, very sensitive.  You removed my comment by the way, was that on purpose?

First thing is, if you want to modify classpath behavior for scaladoc, please do it in doc/Settings.  I don't want any of the main compiler classes touched.  I know too well how easy this is to break.

Second thing is, that's the wrong solution.  It changes a near-universal default without explanation (that no explicit classpath means current directory) and leaves the problem right where it was.  After someone figures out why scaladoc isn't working despite everything being visible from the current directory, they'll pass "-cp ." and be right back where they started.

The question to ask yourself is, why can I run the repl or scalac with "." on my classpath and not see a bunch of spurious messages about "pacckage.class' contains wrong package"? The solution is for scaladoc not to do that, not to futilely try to hide the classfiles from scaladoc.

**Comment by VladUreche** [comment]
Hi Paul,

For the first point:
I just removed a duplicate piece of code in PathResolvers (line 197), the same that can be found in ScalaSettings (line 23, function defaultClasspath). Basically, applied your solution exactly, by adding fallbackClasspath as "." and overriding it in doc.Settings. Nothing more.

For the second point:
I agree that the messages are spurious and shouldn't appear even with the classpath set to ".". I will come up with a better solution.

Bottom line:
I will come with a better solution for the error, but while we're at it, please remove the duplicate code in PathResolvers, it's place is in ScalaSettings.
